### PR TITLE
ci: updated shell script to run separate test

### PR DESCRIPTION
**Pull request by Madhu-1**
Earlier we were running all the linter for non-go files in one short, this won't be helpful for the users who want to run particular tests.

Fixes: #979

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
> Typo in the subjects 'indivisual'.
> 
> The commit message is not very clear... Is this not just renaming the `make go-lint-text` target to `make lint-extras`?

its doing more than that, I have separated each test into separate commands.

File: scripts/lint-extras.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#! /bin/bash
```
Comment (nixpanic) [inline scripts/lint-extras.sh:1]: No space allowed between `#! /executable`

File: scripts/lint-extras.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# vim: set ts=4 sw=4 et :
+
```
Comment (nixpanic) [inline scripts/lint-extras.sh:3]: Add a description of the script in the header. Also include some example of how the script can be executed.

File: scripts/lint-extras.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+set -e
+
+# Run checks from root of the repo
+scriptdir="$(dirname "$(realpath "$0")")"
+cd "$scriptdir/.."
+
+# run_check <file_regex> <checker_exe> [optional args to checker...]
+# Pass empty regex when no regex is needed
+function run_check() {
+    regex="$1"
```
Comment (nixpanic) [inline scripts/lint-extras.sh:13]: this is a bash script, use `local` for variables used only in the function

File: scripts/lint-extras.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    shift
+    exe="$1"
+    shift
+
+    if [ -x "$(command -v "$exe")" ]; then
+        if [ -z "$regex" ]; then
```
Comment (nixpanic) [inline scripts/lint-extras.sh:19]: use `${regex}` notations instead `$regex`, everywhere

File: scripts/lint-extras.sh [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            "$exe" "$@"
+        else
+            find . -path ./vendor -prune -o -regextype egrep -iregex "$regex" -print0 |
```
Comment (nixpanic) [inline scripts/lint-extras.sh:22]: maybe use `git ls-files --exclude='vendor/*' "${regex}"` instead of `find`

Comment (Madhu-1) [reply scripts/lint-extras.sh:25]: this is not working, i will leave it as it is for now as this is copy-pasted from old script file

**Pull request merged**